hs_app_timeseries/receivers.py

```diff
@@ -29,36 +27,37 @@ def resource_pre_create_handler(sender, **kwargs):
         if file_ext not in TimeSeriesResource.get_supported_upload_file_types():
             validate_files_dict['are_files_valid'] = False
             validate_files_dict['message'] = 'Invalid file type.'
-        # if it is a sqlite file validate that it is a valid odm2 db file
-        # if file_ext == '.sqlite':
-        #     err_message = _validate_odm2_db_file(uploaded_file)
-        #     if err_message:
-        #         validate_files_dict['are_files_valid'] = False
-        #         validate_files_dict['message'] = err_message
-
-# check the file to be added is valid
+
+
 @receiver(pre_add_files_to_resource, sender=TimeSeriesResource)
 def pre_add_files_to_resource_handler(sender, **kwargs):
-    # TODO: implement
-    pass
+    resource = kwargs['resource']
+    if resource.files.all().count() > 0:
+        raise ValidationError("Resource is not allowed to have more than 1 content file.")
+
+    files = kwargs['files']
+    validate_files_dict = kwargs['validate_files']
+    if files:
+        infile = files[0]
+        fl_ext = os.path.splitext(infile.file.name)[1]
+        if fl_ext != '.sqlite':
+            validate_files_dict['are_files_valid'] = False
+            validate_files_dict['message'] = "{sql_file} is not a sqlite file".format(sql_file=infile.file.name)
```

Comment:
`get_supported_upload_file_types()` defines the supported types already, which raises two questions:

- Why isn't there some mechanism validating these extensions already for *all* resource types and not specifically implemented for a single one?
- Why is this hardcoding the ".sqlite" string instead of using `get_supported_upload_file_types()`?